services/wallet/router/router_v2.go

```diff
@@ -530,7 +548,23 @@ func (r *Router) SuggestedRoutesV2(ctx context.Context, input *RouteInputParams)
 		}
 	}
 
-	return suggestedRoutes, err
+	mapError := func(err error) error {
+		if err == nil {
+			return nil
+		}
+		pattern := "insufficient funds for gas * price + value: address "
+		addressIndex := strings.Index(err.Error(), pattern)
+		if addressIndex != -1 {
+			addressIndex += len(pattern) + 42
```

Comment:
please define some const like "hexAddressLength" or at least add a comment.
Also, please add a test case covering this.

Reply:
Tests added.